src/DynamoCoreWpf/UI/Prompts/DynamoMessageBox.xaml.cs

```diff
@@ -159,7 +175,24 @@ public static MessageBoxResult Show(string messageBoxText, string caption)
                 BodyText = messageBoxText
             };
             dynamoMessageBox.ConfigureButtons(MessageBoxButton.OK);
-            return dynamoMessageBox.ConvertResult(dynamoMessageBox.ShowDialog());
+            dynamoMessageBox.ShowDialog();
+            return dynamoMessageBox.CustomDialogResult;
+        }
+
+        internal static MessageBoxResult Show(string messageBoxText, string caption, MessageBoxButton button, IEnumerable<string> buttonNames,
```

Comment:
how are the `buttonNames` assigned to the window buttons ?
Is there an order defined somewhere ?
Maybe it should be a list<tuple<button_enum, name>> or dictionary<button_enum, name>

Reply:
as described in the summary, they order just matches the order defined in the ENUM - so OKCancel , "1" ,"2" -> OK:1 Cancel:2.

Reply:
Is that acceptable or do you prefer the more explicit parameter?

Reply:
must of missed the comment, just goes to  show that it is not idiot proof :)
I think it is acceptable for this API (probably not a lot of traffiic)